### PR TITLE
Work Files: Preserve subversion comment of current filename by default

### PR DESCRIPTION
**Pull request by BigRoy**
## Brief description

This is an implementation for #2626 where clicking on Save in the Work Files tool the subversion field will be prefilled with what is assumed to be the current 'comment' in the filename. 

It uses regex so is always prone to error I suppose. Howevr this case should be a pretty accurate detection against the filename template because all other "data" of the current context is formatted like also used for making a new save version- then only the {comment} and {version} are parsed against the current filename.

If the regex cannot detect any comment or current filename can't be matched against the template then nothing changes. Comment field will remain empty.

## Testing notes:
1. try and break the logic
2. test against some unique filename templates used by OP teams

### CODE REVIEW

**Comment by BigRoy** [comment]
With https://github.com/pypeclub/OpenPype/pull/2734/commits/18f8741224d06e6aee41c924a9ad7796810072cd this now also implements #2719 

Admittedly the code might benefit from some cleanup - moving widgets to a `widgets.py` or alike. But it's functional at least. :)

File: openpype/tools/workfiles/app.py [hunk 1]
```diff
@@ -37,6 +38,182 @@
...
+        if not self.fname_regex:
+            return
+
+        fname = os.path.basename(filepath)
+        match = self.fname_regex.match(fname)
+        if match:
+            return match.group(1)
+
+
+class SubversionLineEdit(QtWidgets.QWidget):
```
Comment (BigRoy) [inline openpype/tools/workfiles/app.py:151]: This is basically similar to the "subset" line edit as in the Creator tool.
Wasn't sure how to implement this simple widget with less code than this.

File: openpype/tools/workfiles/app.py [hunk 2]
```diff
@@ -37,6 +38,182 @@
...
+        menu = self._menu
+        button = self._button
+
+        state = any(values)
+        button.setEnabled(state)
+        if state is False:
+            return
+
+        # Include an empty string
+        values = [""] + sorted(values)
```
Comment (BigRoy) [inline openpype/tools/workfiles/app.py:201]: Empty string is just there so you can also quickly clear the field when checking the dropdown menu.

**Comment by antirotor** [comment]
It doesn't get this:
image

I thought there should be `test` in subversion textfield?

**Comment by BigRoy** [comment]
It should indeed!

What is your project's file template?

**Comment by BigRoy** [comment]
Ok. Found it. It's a Py2 <-> Py3 difference in how regex escapes.

In Python 2 a lot more characters are actually being escaped with `re.escape()`. I've fixed the logic.
Can you try again @antirotor?

**Review by antirotor — APPROVED** [review]
That did the trick! Working fine.

**Pull request merged**